fuse_publishers/test/test_pose_2d_publisher.cpp

```diff
@@ -143,85 +144,105 @@ class Pose2DPublisherTestFixture : public ::testing::Test
+    pose_with_covariance_msg_ = msg;
   }
 
-  void tfCallback(const tf2_msgs::TFMessage::ConstPtr& msg)
+  void tfCallback(const tf2_msgs::msg::TFMessage& msg)
   {
     received_tf_msg_ = true;
-    tf_msg_ = *msg;
+    tf_msg_ = msg;
   }
 
+   std::thread spinner_;  //!< Internal thread for spinning the executor
+   rclcpp::executors::SingleThreadedExecutor::SharedPtr executor_;
+
 protected:
-  // TODO(CH3): Replace with node_ (also, we don't need to support node interfaces here since it's a test...)
-  ros::NodeHandle node_handle_;
-  ros::NodeHandle private_node_handle_;
+  geometry_msgs::msg::TransformStamped odom_to_base_;
   fuse_graphs::HashGraph::SharedPtr graph_;
   fuse_core::Transaction::SharedPtr transaction_;
   bool received_pose_msg_;
-  geometry_msgs::PoseStamped pose_msg_;
+  geometry_msgs::msg::PoseStamped pose_msg_;
   bool received_pose_with_covariance_msg_;
-  geometry_msgs::PoseWithCovarianceStamped pose_with_covariance_msg_;
+  geometry_msgs::msg::PoseWithCovarianceStamped pose_with_covariance_msg_;
   bool received_tf_msg_;
-  tf2_msgs::TFMessage tf_msg_;
-  tf2_ros::StaticTransformBroadcaster static_broadcaster_;
+  tf2_msgs::msg::TFMessage tf_msg_;
 };
 
 TEST_F(Pose2DPublisherTestFixture, PublishPose)
 {
   // Test that the expected PoseStamped message is published
+  rclcpp::NodeOptions options;
+  options.arguments({
+    "--ros-args",
+    "-p", "map_frame:=test_map",
+    "-p", "odom_frame:=test_odom",
+    "-p", "base_frame:=test_base",
+    "-p", "publish_to_tf:=false"});
+  auto node = rclcpp::Node::make_shared("test_pose_2d_publisher_node", options);
+  executor_->add_node(node);
 
   // Create a publisher and send it the graph
-  private_node_handle_.setParam("test_publisher/map_frame", "test_map");
-  private_node_handle_.setParam("test_publisher/odom_frame", "test_odom");
-  private_node_handle_.setParam("test_publisher/base_frame", "test_base");
-  private_node_handle_.setParam("test_publisher/publish_to_tf", false);
   fuse_publishers::Pose2DPublisher publisher;
-  publisher.initialize("test_publisher");
+  publisher.initialize(node, "test_publisher");
   publisher.start();
 
   // Subscribe to the "pose" topic
-  ros::Subscriber subscriber = private_node_handle_.subscribe(
-    "test_publisher/pose",
-    1,
-    &Pose2DPublisherTestFixture::poseCallback,
-    reinterpret_cast<Pose2DPublisherTestFixture*>(this));
+  auto subscriber1 = node->create_subscription<geometry_msgs::msg::PoseStamped>(
+    "/test_publisher/pose", 1,
+    std::bind(&Pose2DPublisherTestFixture::poseCallback, this, std::placeholders::_1));
 
   // Send the graph to the Publisher to trigger message publishing
   publisher.notify(transaction_, graph_);
 
   // Verify the subscriber received the expected pose
-  rclcpp::Time timeout = node_->now() + rclcpp::Duration::from_seconds(10.0);
-  while ((!received_pose_msg_) && (node_->now() < timeout))
+  rclcpp::Time timeout = node->now() + rclcpp::Duration::from_seconds(20.0);
```

Comment:
why was the timeout increased from 10s to 20s?

Reply:
I found that it sometimes times out if left at 10 seconds, same with the ones below

Reply:
Interesting! I wonder if the timeout was happening because of the extra 5 second sleep that was removed https://github.com/locusrobotics/fuse/pull/299#discussion_r1054773194

I changed the timeouts back to 10 seconds and didn't see any failures in 100 runs using:

```
colcon test --event-handlers console_direct+ --packages-select fuse_publishers --ctest-args -R _2d_publisher --retest-until-fail 100
```

https://github.com/methylDragon/fuse/pull/4